tests/xharness/Logging/ILog.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.IO;
+using System.Text;
+
+namespace xharness.Logging {
+
+	// defines common log types
+	public enum LogType {
+		XML_LOG,
+		NUNIT_RESULT,
+		SYSTEM_LOG,
+		COMPANION_SYSTEM_LOG,
+		BUILD_LOG,
+		TEST_LOG,
+		EXTENSION_TEST_LOG,
+		EXECUTION_LOG,
+	}
+
+	public static class LogTypeExtensions {
+
+		public static string ToString (this LogType type)
+		{
+			switch (type) {
+			case LogType.XML_LOG:
+				return "XML log";
+			case LogType.NUNIT_RESULT:
+				return "NUnit results";
+			case LogType.SYSTEM_LOG:
+				return "System log";
+			case LogType.COMPANION_SYSTEM_LOG:
+				return "System log (companion)";
+			case LogType.BUILD_LOG:
+				return "Build log";
+			case LogType.TEST_LOG:
+				return "Test log";
+			case LogType.EXTENSION_TEST_LOG:
+				return "Extension test log";
+			case LogType.EXECUTION_LOG:
+				return "Execution log";
+			default:
+				throw new ArgumentException ($"Unknoe tyme for {nameof (type)}");
+			}
+		}
+	}
+	public interface ILog : IDisposable {
+		ILogs Logs { get; }
+		string Description { get; set; }
```

Comment:
>string [](start = 2, length = 6)

Based on the usage I've seen, this also might be `LogType` and we could get the Description from that always